ddl/ddl.go

```diff
@@ -121,16 +124,26 @@ func (d *ddl) Stop() error {
 	err := kv.RunInNewTxn(d.store, true, func(txn kv.Transaction) error {
 		t := meta.NewMeta(txn)
 		owner, err := t.GetDDLOwner()
-		if err != nil || owner == nil {
+		if err != nil {
 			return errors.Trace(err)
 		}
-
-		if owner.OwnerID != d.uuid {
+		if owner == nil || owner.OwnerID != d.uuid {
 			return nil
 		}
-
 		// owner is mine, clean it so other servers can compete for it quickly.
-		return errors.Trace(t.SetDDLOwner(&model.Owner{}))
+		if err = t.SetDDLOwner(&model.Owner{}); err != nil {
+			return errors.Trace(err)
+		}
+
+		// clean task owner like job owner.
+		owner, err = t.GetDDLTaskOwner()
```

Comment:
no, it shouldn't. The two owners aren't the same to make it real executed asynchronously.